src/SearchNamespace.js

```diff
@@ -0,0 +1,128 @@
+import GMusicNamespace from './GMusicNamespace';
+
+import Album from './Structs/Album';
+import Artist from './Structs/Artist';
+import Track from './Structs/Track';
+
+import { findContextPath } from './utils/context';
+
+export default class SearchNamespace extends GMusicNamespace {
+  constructor(...args) {
+    super(...args);
+    this.path = findContextPath();
+    if (this.path) {
+      this._watchForSearches();
+    }
+
+    this.addMethod('getSearchText', this.getSearchText.bind(this));
+    this.addMethod('getResults', this.getResults.bind(this));
+    this.addMethod('isSearching', this.isSearching.bind(this));
+    this.addMethod('playResult', this.playResult.bind(this));
+    this.addMethod('search', this.search.bind(this));
+  }
+
+  _text(elem, def) {
+    if (elem) {
+      return elem.textContent.trim();
+    }
+    return def;
+  }
+
+  _watchForSearches() {
+    const that = this;
+    let searchChanged = 0;
+
+    window.addEventListener('hashchange', () => {
+      if (!/^#\/sr\//g.test(window.location.hash)) return;
+      this.emitter.emit('change:search:text', this.getSearchText());
+      searchChanged = 2;
+    });
+
+    window.APPCONTEXT[this.path[0]][this.path[1]][0].addEventListener('E', () => {
+      if (searchChanged > 0) {
+        searchChanged -= 1;
+        if (searchChanged === 0) {
+          that.emitter.emit('change:search:results', that.getResults());
+        }
+      }
+    });
+  }
+
+  getSearchText() {
+    return document.querySelector('sj-search-box input').value;
+  }
+
+  getResults() {
+    if (!this.isSearching()) {
+      throw new Error('Can\'t get search results when the user is not searching');
+    }
+
+    const albumElems = document.querySelectorAll('.lane-content > [data-type=album]');
+    const albums = [];
+
+    Array.prototype.forEach.call(albumElems, (elem) => {
+      albums.push(new Album(
+        elem.getAttribute('data-id'),
+        elem.querySelector('.details .title').textContent,
+        elem.querySelector('.details .sub-title').textContent,
+        elem.querySelector('img').src.replace('=w220-c-h220-e100', '')
+      ));
+    });
+
+    const artistElems = document.querySelectorAll('.lane-content > [data-type=artist]');
+    const artists = [];
+
+    Array.prototype.forEach.call(artistElems, (elem) => {
+      let image = elem.querySelector('img');
+      if (image) {
+        image = image.src.replace('=w220-c-h220-e100', '');
```

Comment:
https://lh3.googleusercontent.com/4Yht2ETGQNme6QgQi-imsOK788OEHEhldqgBjeR8hWi8YsUMbn_AY0c5COHB4wK5C3Hjiw-y3Q=w190-c-h190-e100

They always look like this, the final parameters dictate size and quality.  Removing them force google to send the original image not a scaled or compressed one.  Happy to remove if you don't like it 👍